homeassistant/components/climate/intesishome.py

```diff
@@ -0,0 +1,350 @@
+        self._rssi = self._intesishome.get_rssi(self._deviceid)
+        self._run_hours = self._intesishome.get_run_hours(self._deviceid)
+
+        # Operation mode
+        mode = self._intesishome.get_mode(self._deviceid)
+        if self._intesishome.get_power_state(self._deviceid) == 'off':
+            self._current_operation = STATE_OFF
+            self._fan_speed = None
+            self._swing = None
+        elif mode == 'auto':
+            self._current_operation = STATE_AUTO
+        elif mode == 'fan':
+            self._current_operation = STATE_FAN
+        elif mode == 'heat':
+            self._current_operation = STATE_HEAT
+        elif mode == 'dry':
+            self._current_operation = STATE_DRY
+        elif mode == 'cool':
+            self._current_operation = STATE_COOL
+        else:
+            self._current_operation = STATE_UNKNOWN
+
+        # Target temperature
+        if self._current_operation in [STATE_OFF, STATE_FAN]:
+            self._target_temp = None
+        else:
+            self._target_temp = self._intesishome.get_setpoint(self._deviceid)
+
+        # Fan speed
+        fan_speed = self._intesishome.get_fan_speed(self._deviceid)
+        if fan_speed:
+            # Capitalize fan speed from pyintesishome
+            self._fan_speed = fan_speed[:1].upper() + fan_speed[1:]
+
+        # Swing mode
+        # Climate module only supports one swing setting, so use vertical swing
+        swing = self._intesishome.get_vertical_swing(self._deviceid)
+        if not self._has_swing_control:
+            # Device doesn't support swing
+            self._swing = None
+        elif swing == 'auto/stop':
+            self._swing = "Auto/Stop"
+        elif swing == 'swing':
+            self._swing = "Swing"
+        elif swing == 'manual3':
+            self._swing = "Middle"
+        else:
+            self._swing = STATE_UNKNOWN
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def _poll_status(self, shouldcallback):
+        """Internal method to poll IntesisHome via HTTP."""
+        _LOGGER.debug("Polling IntesisHome Status via HTTP")
+        self._intesishome.poll_status(shouldcallback)
+
+    @property
+    def icon(self):
+        """Set the climate icon to the approprate function."""
+        icon = None
+        if self._current_operation == STATE_HEAT:
+            icon = 'mdi:white-balance-sunny'
+        elif self._current_operation == STATE_FAN:
+            icon = 'mdi:fan'
+        elif self._current_operation == STATE_DRY:
+            icon = 'mdi:water-off'
+        elif self._current_operation == STATE_COOL:
+            icon = 'mdi:nest-thermostat'
+        elif self._current_operation == STATE_AUTO:
+            icon = 'mdi:cached'
+        return icon
+
+    def update_callback(self):
+        """Called when data is received by pyIntesishome."""
+        _LOGGER.debug("IntesisHome sent a status update.")
+        self.hass.async_add_job(self.update_ha_state, True)
```

Comment:
Hi there,

I have a lot of deprecation warning assuming caused by this line.

`WARNING (Thread-5) [homeassistant.helpers.entity] 'update_ha_state' is deprecated. Use 'schedule_update_ha_state' instead.`